src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -622,7 +622,7 @@ public void SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, bool pollBef
                     }
                 });
 
-                Assert.True(Task.WaitAll(new[] { serverTask, clientTask }, TestTimeout), "Wait timed out");
+                Assert.True(await (new[] { serverTask, clientTask }).WhenAllWithTimeout(TestTimeout), "Wait timed out");
```

Comment:
An extension method on an array of tasks is odd to me personally.  I'd find it easier to understand if it were called with the normal static method syntax.